python2/koans/about_classes.py

```diff
@@ -125,36 +129,36 @@ def get_self(self):
             return self
 
         def __str__(self):
-            #
-            # Implement this!
-            #
-            return __
+            name = str(self._name)
+            return name
 
         def __repr__(self):
             return "<Dog named '" + self._name + "'>"
 
     def test_inside_a_method_self_refers_to_the_containing_object(self):
         fido = self.Dog6("Fido")
 
-        self.assertEqual(__, fido.get_self())  # Not a string!
+        self.assertEqual(fido, fido.get_self())  # Not a string!
 
     def test_str_provides_a_string_version_of_the_object(self):
         fido = self.Dog6("Fido")
         self.assertEqual("Fido", str(fido))
 
     def test_str_is_used_explicitly_in_string_interpolation(self):
         fido = self.Dog6("Fido")
-        self.assertEqual(__, "My dog is " + str(fido))
+        self.assertEqual("My dog is Fido", "My dog is " + str(fido))
 
     def test_repr_provides_a_more_complete_string_version(self):
         fido = self.Dog6("Fido")
-        self.assertEqual(__, repr(fido))
+        self.assertEqual("<Dog named 'Fido'>", repr(fido))
 
     def test_all_objects_support_str_and_repr(self):
         seq = [1, 2, 3]
 
-        self.assertEqual(__, str(seq))
-        self.assertEqual(__, repr(seq))
+        self.assertEqual('[1, 2, 3]', str(seq))
+        self.assertEqual('[1, 2, 3]', repr(seq))
+
+        self.assertEqual('STRING', str("STRING"))
+        self.assertEqual("'STRING'", repr("STRING"))
 
-        self.assertEqual(__, str("STRING"))
-        self.assertEqual(__, repr("STRING"))
+#In the last one I'm not sure why the "" appears twice
```

Comment:
[This page](https://www.geeksforgeeks.org/str-vs-repr-in-python/) goes into the differences between `str` and `repr`. Give it a read and let me know if it clear things for you. If not, we can discuss about it.

Reply:
I understand it, but we can discuss it next to see if what I understand it's correct, haha.